### ISSUE TITLE
log_driver defaults to json-file instead of letting system decide

### ISSUE THREAD

**Issue by stephenliberty** [kreuzwerker/terraform-provider-docker#269]
Because log_driver is defaulted, it ignores what the system's default is. So, as an example, I've got my docker daemon setup to send longs to journald. But nothing logs out because the provider is hard set to use json-file. 

The idea that I'd have to hard set on every container what logging mechanism to use seems.. not great. Especially since this TF may be applied against different machines with different logging setups. 

### Affected Resource(s)

* `docker_container`

### Terraform Configuration Files
```hcl
resource "docker_container" "nginx" {
  name = "nginx"
  image = "nginx:latest"
}
```

/etc/docker/daemon.json
```json
{
  "log-driver": "journald"
}
```

### Debug Output

```
  + resource "docker_container" "nginx" {
      + attach           = false
      + bridge           = (known after apply)
      + command          = (known after apply)
      + container_logs   = (known after apply)
      + entrypoint       = (known after apply)
      + env              = (known after apply)
      + exit_code        = (known after apply)
      + gateway          = (known after apply)
      + hostname         = (known after apply)
      + id               = (known after apply)
      + image            = "nginx:latest"
      + init             = (known after apply)
      + ip_address       = (known after apply)
      + ip_prefix_length = (known after apply)
      + ipc_mode         = (known after apply)
      + log_driver       = "json-file"
      + logs             = false
      + must_run         = true
      + name             = "nginx"
      + network_data     = (known after apply)
      + read_only        = false
      + remove_volumes   = true
      + restart          = "no"
      + rm               = false
      + security_opts    = (known after apply)
      + shm_size         = (known after apply)
      + start            = true
      + stdin_open       = false
      + tty              = false

      + healthcheck {
          + interval     = (known after apply)
          + retries      = (known after apply)
          + start_period = (known after apply)
          + test         = (known after apply)
          + timeout      = (known after apply)
        }

      + labels {
          + label = (known after apply)
          + value = (known after apply)
        }
    }
```

### Expected Behaviour

Either the `log_driver` should be empty, or set to the same as what the system is set to. 

### Actual Behaviour

The `log_driver` is set to `json-file`

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Docker daemon set to use journald logging.

**Comment by suzuki-shunsuke** [comment]
This is same as https://github.com/hashicorp/terraform-provider-docker/issues/270 and I fixed this in https://github.com/hashicorp/terraform-provider-docker/pull/269 .
I removed the default value of log_driver, but I found the default value is set currently.

I checked the Git history.

* https://github.com/hashicorp/terraform-provider-docker/commit/cdc0d912cc7f44400e587a997594e83688dc0430#diff-5534a8bf4e2b1faee1078a46cbbbcd5eff4dd5b3b787f157189477980ae93955R614

**Comment by suzuki-shunsuke** [comment]
https://github.com/kreuzwerker/terraform-provider-docker/pull/270

**Comment by suzuki-shunsuke** [comment]
📝 

This problem has been fixed in v2.7.1 once, but it has degraded in v2.8.0.

* [v2.7.0](https://github.com/hashicorp/terraform-provider-docker/blob/v2.7.0/docker/resource_docker_container.go#L602)
* https://github.com/hashicorp/terraform-provider-docker/pull/269
* [v2.7.1](https://github.com/hashicorp/terraform-provider-docker/blob/v2.7.1/docker/resource_docker_container.go#L603)
* [v2.7.2](https://github.com/hashicorp/terraform-provider-docker/blob/v2.7.2/docker/resource_docker_container.go#L604)
* https://github.com/hashicorp/terraform-provider-docker/commit/cdc0d912cc7f44400e587a997594e83688dc0430#diff-5534a8bf4e2b1faee1078a46cbbbcd5eff4dd5b3b787f157189477980ae93955R614
* [v2.8.0](https://github.com/kreuzwerker/terraform-provider-docker/blob/v2.8.0/docker/resource_docker_container.go#L619)
